src/lib/services/transfer/__tests__/__snapshots__/access.spec.ts.snap

```diff
@@ -0,0 +1,84 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`access create 1`] = `
+{
+  "headers": {
+    "accept": "*/*",
+    "accept-encoding": "gzip,deflate",
+    "connection": "close",
+    "content-length": "59",
+    "content-type": "application/json",
+    "host": "transfer.api.globusonline.org",
+    "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
+  },
+  "json": {
+    "path": "/",
```

Comment:
I'm not actually sure what a minimal payload would be on this. The type is just a partial access record.

Reply:
The `Partial<...>` usage there actually seems incorrect; We know `DATA_TYPE` is required, and many of our other methods will inject that property as a convenience (this one doesn't). There is also the `role_*` vs `principal` usage... keeping it loose (`Partial<..>`) for now, is ok, but once we start using it we should probably come back and make this more strict.

```ts
  interface AccessDocument extends TransferEntity {
    DATA_TYPE: `${DATA_TYPE.access}`;
    id: null | Globus.UUID;
    role_id: null | Globus.UUID;
    role_type:
      | null
      | 'administrator'
      | 'access_manager'
    principal_type:
      | 'identity'
      | 'group'
      | 'all_authenticated_users'
      | 'anonymous';
    principal: '' | Globus.UUID;
    path: string;
    permissions: 'r' | 'rw';
    create_time: null | string;
    notify_email?: null | string;
    notify_message?: null | string;
  }
```
> The notify_email and notify_message fields are optional and not persisted as part of the access document.

> The id field of the access document must be omitted in create requests.

https://docs.globus.org/api/transfer/acl/#rest_access_create

It might actually wind up being something like:

```ts
type AccessDocumentCreate = Omit<AccessDocument, 'id' | 'role_id' | 'create_time'> & ( { role_type: ... } | { principal: ... } | { principal_type: ...} ); // Not sure if that actually describes the permutations, but hopefully it communicates the idea.
```